### PR TITLE
Bring opds registration up to spec

### PR DESCRIPTION
**Pull request by aslagle**

### CODE REVIEW

File: controller.py [hunk 1]
```diff
@@ -168,6 +170,22 @@ def register(self, do_get=HTTP.get_with_timeout):
         SHELF_REL = "http://opds-spec.org/shelf"
 
         auth_response = None
+        def get_opds_links(response):
+            type = response.headers.get("Content-Type")
+            if type == "application/opds+json":
```
Comment (leonardr) [inline controller.py]: Why no startswith here?

Comment (aslagle) [inline controller.py]: OPDS 2 doesn't have different kinds of feeds like OPDS 1, so it's not necessary. I guess it wouldn't hurt to have it anyway.

Comment (leonardr) [inline controller.py]: It does, because of `profile`, but right now the only profile is the directory one, and we don't want to import another directory.

Comment (aslagle) [inline controller.py]: In that case maybe it shouldn't have startswith.

Comment (leonardr) [inline controller.py]: Yeah, not for now, certainly.

**Pull request merged**